src/particles/deposition/TemperatureDeposition.cpp

```diff
@@ -0,0 +1,161 @@
+        // extract laser properties and boolean for the presence of the laser and for ionization
+        const PhysConst pc = get_phys_const();
+        const int aabs = Hipace::m_use_laser ? Comps[WhichSlice::This]["aabs"] : -1;
+        const amrex::Real clightinv = 1.0_rt/pc.c;
+        const bool can_ionize = plasma.m_can_ionize;
+        const bool use_laser = Hipace::m_use_laser;
+        const amrex::Real laser_norm = (plasma.m_charge/pc.q_e) * (pc.m_e/plasma.m_mass)
+            * (plasma.m_charge/pc.q_e) * (pc.m_e/plasma.m_mass);
+
+        // Loop over particles
+        SharedMemoryDeposition<1, 1, true>(
+            int(pti.numParticles()),
+            // is_valid
+            // return whether the particle is valid and should deposit
+            [=] AMREX_GPU_DEVICE (int ip, auto ptd)
+            {
+            // only deposit on or below their according MR level
+                return ptd.id(ip).is_valid() && (lev == 0 || ptd.cpu(ip) >= lev);
+            },
+            [=] AMREX_GPU_DEVICE (int ip, auto ptd) -> amrex::IntVectND<2>
+            {
+                const amrex::Real xp = ptd.pos(0, ip);
+                const amrex::Real yp = ptd.pos(1, ip);
+
+                const amrex::Real xmid = (xp - x_pos_offset) * dx_inv;
+                const amrex::Real ymid = (yp - y_pos_offset) * dy_inv;
+
+                auto [shape_x, i] =
+                compute_single_shape_factor<false, 0>(xmid, 0);
+
+                auto [shape_y, j] =
+                compute_single_shape_factor<false, 0>(ymid, 0);
+
+                return {i, j};
+            },
+            // deposit of weight, momentum (ux, uy, uz) and their squares (uxsq, uysq, uzsq)
+            [=] AMREX_GPU_DEVICE (int ip, auto ptd,
+                                  Array3<amrex::Real> arr,
+                                  auto cache_idx, auto depos_idx) noexcept
+            {
+                const amrex::Real xp = ptd.pos(0, ip);
+                const amrex::Real yp = ptd.pos(1, ip);
+
+                amrex::Real Aabssqp = 0._rt;
+                amrex::Real laser_norm_ion = laser_norm;
+                if (can_ionize) {
+                    laser_norm_ion *=
+                        ptd.idata(PlasmaIdx::ion_lev)[ip] * ptd.idata(PlasmaIdx::ion_lev)[ip];
+                }
+                if (use_laser) {
+                    doLaserGatherShapeN<0>(xp, yp, Aabssqp, arr, cache_idx[0],
+                                                    dx_inv, dy_inv, x_pos_offset, y_pos_offset);
+                    Aabssqp *= laser_norm_ion;
+                }
+
+                const amrex::Real uxp = ptd.rdata(PlasmaIdx::ux)[ip]*clightinv;
+                const amrex::Real uyp = ptd.rdata(PlasmaIdx::uy)[ip]*clightinv;
+                amrex::Real psi = ptd.rdata(PlasmaIdx::psi)[ip];
+                const amrex::Real uzp = (1._rt + uxp*uxp + uyp*uyp
+                    + 0.5_rt*Aabssqp*0. - psi*psi)/(2.*psi);
```

Comment:
```suggestion
                const amrex::Real uzp = (1._rt + uxp*uxp + uyp*uyp
                    + 0.5_rt*Aabssqp - psi*psi)/(2.*psi);
```
Is this more correct? In any case some fix seems to be needed there.

Reply:
Yes exactly, it will be fixed in an other pull request. Because we need also to change it for the insitu-diagnostics to make the CI test work correctly. So, it is just temporarily